### DISCUSSION TITLE
Java Application crash after create data writer or create data reader call

### DISCUSSION THREAD

**Discussion by kubraturhan** [OpenDDS/OpenDDS#4476] [Q&A]
After I successfully created a topic via create_topic method of DomainParticipantTaoPeer, whenever I call create_datawriter or create_datareader method of PublisherTaoPeer the java application is crashing. I share the code and the error logs. Can you help me please?

![StackTrace](https://github.com/OpenDDS/OpenDDS/assets/3930719/e834c371-634b-4752-aecb-2cae9f9c833f)
![Exception](https://github.com/OpenDDS/OpenDDS/assets/3930719/87c9ae84-f97e-4d1f-a7ff-cfcf8f328cd2)
![code](https://github.com/OpenDDS/OpenDDS/assets/3930719/117e83de-0689-4b24-8532-d82e53f84025)

**Comment by mitza-oci** [comment]
Please share the IDL used to compile SI_UY_Arayuz.dll.  Share it as text, either directly in this discussion, as a file attachment, or in a GitHub repository.  Please do not send an image.

**Reply by kubraturhan** [reply]
After a long time of investigation, I am sure that idl file format is correct. IDL file is a big file containing 980 lines, 53 topics, 48 enums and 80 structs nearly 21KB. I divide the file into two parts and each one worked correctly. But if I use the file as a whole it didnt work. I think OpenDDS have a limit for IDL file. We upgraded OpenDDS to 3.26 version. While working with OpenDDS 3.11 version with the same IDL file there is not any problem. Do you have any limitation defined for OpenDDS IDL files?

**Reply by mitza-oci** [reply]
I was not suggesting that the IDL format was not correct, but it's difficult for me to analyze this in depth without having the IDL available.
I think it's clear that some limit is being reached but it would be good to understand more about it.
The error is a stack overflow, so you can try increasing the stack.
